README.md

```diff
@@ -58,6 +58,20 @@ $ curl http://0.0.0.0:8080/record/23
 ## Next.js app
 When you run for the first time, you will need to copy the template .env.example file and add missed variables:
 
-   ```sh
-   $ cp .env.example .env
-   ```
\ No newline at end of file
+```sh
+$ cp .env.example .env
+```
+
+I've added `docker-compose-dev` - use this file if you need to make changes to the code. Next.js will compile into a dev build automatically after any changes are made to the JS (Type Script) code. This build is convenient for development, but it's about 30-40% slower than the "production" build. Therefore, if you want to get maximum performance, use:
+```
+$ docker-compose build 
+$ docker-compose up
+```
+If you want the ability to change the code and see changes immediately, use:
+
+```
+$  docker-compose -f docker-compose-dev.yml  build
+$  docker-compose -f docker-compose-dev.yml  up
+```
+
+For deployment, use only prod Dockerfile (not Dockerfile.dev). Note that it should receive `ARG EXTERNAL_API` to work correctly with the Python API.
```

Comment:
Is it not possible to set this as an environmental variable, rather than passing it as an argument to the docker build command? This feels like a problematic workaround - is there some specific reason why we should do this?

Reply:
The reason we need to pass this as an argument is that we are building the project during the creation of the Docker image to make an optimized production build. Docker compose does not pass the values from environment variables during the image build. This is the reason you encountered difficulties the first time. The `NEXT_PUBLIC_API_URL=http://localhost:8080` simply wasn't passed at the correct stage.

Reply:
Got it, thanks!